Mathlib/Algebra/MonoidAlgebra/Basic.lean

```diff
@@ -960,6 +960,15 @@ def mapDomainAlgHom (k A : Type*) [CommSemiring k] [Semiring A] [Algebra k A] {H
 #align monoid_algebra.map_domain_alg_hom MonoidAlgebra.mapDomainAlgHom
 #align monoid_algebra.map_domain_alg_hom_apply MonoidAlgebra.mapDomainAlgHom_apply
 
+@[simps!]
+def mapDomainAlgEquiv (k A : Type*) [CommSemiring k] [Semiring A] [Algebra k A] {H : Type*}
+    [Monoid H] (e : G ≃* H) :
+    MonoidAlgebra A G ≃ₐ[k] MonoidAlgebra A H :=
+  { mapDomainAlgHom k A e with
+    invFun := mapDomainAlgHom k A e.symm
+    left_inv := fun f ↦ by simp [← Finsupp.mapDomain_comp]
+    right_inv := fun f ↦ by simp [← Finsupp.mapDomain_comp] }
```

Comment:
style nit:
```suggestion
    MonoidAlgebra A G ≃ₐ[k] MonoidAlgebra A H where
  __ := mapDomainAlgHom k A
  invFun := mapDomainAlgHom k A e.symm
  left_inv f := by simp [← Finsupp.mapDomain_comp]
  right_inv f := by simp [← Finsupp.mapDomain_comp]
```

Reply:
I can't get this to work!

Reply:
I've realized that this actually has the wrong defeqs, and we instead want to wrap `Finsupp.domLCongr`. I'll make a separate PR.

Reply:
#6567